Block/Embed.php

```diff
@@ -160,12 +175,45 @@ public function getCurrentCustomerDetails()
         }
 
         $customerSession = $this->modelSessionFactory->getCustomer();
+
+        try {
+            $jsApiKey = $this->decryptJsApiKey($this->model->getJsApiKey());
+            $hash = hash_hmac('sha256', $customerSession->getEmail(), $jsApiKey);
+        } catch (\Exception $e) {
+            $hash = '';
+        }
+
         return [
             'name'  => $customerSession->getName(),
-            'email' => $customerSession->getEmail()
+            'email' => $customerSession->getEmail(),
+            'hash' => $hash
         ];
     }
 
+    /**
+     * Retrieve JS API key
+     *
+     * @param string $js_api_key Encrypted JS API key
+     * @return string
+     * @throws \Exception error retrieving JS API key
+     */
+    private function decryptJsApiKey(string $js_api_key)
+    {
+        if (empty($js_api_key)) {
+            throw new LocalizedException(__('JS API key is empty'));
+        }
+
+        if ($this->modelSessionFactory->hasData(self::TAWKTO_JS_API_KEY)) {
+            return $this->modelSessionFactory->getData(self::TAWKTO_JS_API_KEY);
```

Comment:
Same as with wordpress update - if modelSessionFactory is a client session - we should not cache the API key in it, just the hashed value